src/libaktualizr/uptane/uptane_test.cc

```diff
@@ -459,57 +461,58 @@ TEST(Uptane, InstallFakeBad) {
 
   std::string hash = download_result.updates[0].sha256Hash();
   std::transform(hash.begin(), hash.end(), hash.begin(), ::toupper);
-  boost::filesystem::path image = temp_dir / "images" / hash;
+  auto image = (conf.pacman.images_path / hash).string();
 
   // Overwrite the file on disk with garbage so that the target verification
   // fails. First read the existing data so we can re-write it later.
-  auto rhandle = storage->openTargetFile(download_result.updates[0]);
   const uint64_t length = download_result.updates[0].length();
-  uint8_t content[length];
-  EXPECT_EQ(rhandle->rread(content, length), length);
-  rhandle->rclose();
-  auto whandle = storage->allocateTargetFile(download_result.updates[0]);
-  uint8_t content_bad[length + 1];
+  char content[length];
+  auto r = std::ifstream(image, std::ios::binary);
+  r.read(content, length);
+  EXPECT_EQ(r.gcount(), length);
+  r.close();
+  auto w = std::ofstream(image, std::ios::binary | std::ios::ate);
+  char content_bad[length + 1];
   memset(content_bad, 0, length + 1);
-  EXPECT_EQ(whandle->wfeed(content_bad, 3), 3);
-  whandle->wcommit();
+  w.write(content_bad, 3);
+  w.close();
```

Comment:
This test is such a hack but it does the job. :) It also conveniently tested a few of the old Storage APIs, though. Do we have good enough coverage of the new versions?

Reply:
It should be covered by new unit tests in packagemanagerfake and storage.